### ISSUE TITLE
Unable to find srcml.h with registered srcML in CMake when installed in non-default location

### ISSUE THREAD

**Issue by mlcollard** [srcML/srcML#1936]
On some platforms, specifically Windows, srcML can be installed to non-default locations, e.g., instead of C:\Program Files\srcML it can be installed to B:\srcML. When installed on Windows, the installer registers srcML with CMake so that any CMake build will know about srcML and can use `find_package(srcML)` to use it in their application.

When installed to the default location, e.g., C:\Program Files\srcML, applications get the include file location and are able to build. However, when installed to a non-default location, e.g., B:\srcML, they have a missing include file.

This happens the most on Windows because on Linux and macOS the include files go into a common directory.

**Comment by mlcollard** [comment]
Appears the problem is this cmake build code:

```cmake
find_path(srcML_INCLUDE_DIR
  NAMES srcml.h
  PATHS "${CMAKE_CURRENT_LIST_DIR}/../include"
  DOC "libsrcml include directory"
)
```

The `CMAKE_CURRENT_LIST_DIR` is the location of the srcMLConfig file. This is in the root of the installed srcML. The reason it worked for the default install is that CMake knew enough to look for it in srcML/include. However, with the non-default install, the standard locations did not work, and the path given above is incorrect.

Will fix to the following:

```cmake
find_path(srcML_INCLUDE_DIR
  NAMES srcml.h
  PATHS "${CMAKE_CURRENT_LIST_DIR}/../../../include"
  NO_DEFAULT_PATH
  DOC "libsrcml include directory"
)
```

Note that the `NO_DEFAULT_PATH` is useful for debugging, since then it fails everywhere if the path is not correct. Will remove the `NO_DEFAULT_PATH` for distribution.

**Comment by mlcollard** [comment]
Merged into develop in #1940
